### PR TITLE
make system tests runnable on non kind clusters

### PR DESCRIPTION
**Pull request by aclevername**
Changes
- Expose environment variables for cluster kubeconfigs
- Pushed the bash images to dockerhub
- Use an in-cluster pod to handle the promiserelease curls
- Skip checking the minio bucket. I almost wanted to delete this test and didn't, I don't think its worth making it work on any s3 provider

### CODE REVIEW

File: test/system/system_suite_test.go [hunk 1]
```diff
@@ -29,17 +28,63 @@ func TestSystem(t *testing.T) {
...
+		checkExitCode: true,
+		name:          getEnvOrDefault("WORKER_NAME", "worker-1"),
+	}
+	platform = &destination{
+		context:       getEnvOrDefault("PLATFORM_CONTEXT", "kind-platform"),
+		checkExitCode: true,
+		name:          getEnvOrDefault("PLATFORM_NAME", "platform-cluster"),
+	}
+
+	endpoint = getEnvOrDefault("BUCKET_ENDPOINT)", "localhost:31337")
```
Comment (kirederik) [inline test/system/system_suite_test.go:74]: `BUCKET_ENDPOINT)` seems to have an extra `)`

File: test/system/system_suite_test.go [hunk 2]
```diff
@@ -29,17 +28,63 @@ func TestSystem(t *testing.T) {
...
+	secretAccessKey = getEnvOrDefault("BUCKET_SECRET_KEY", "minioadmin")
+	accessKeyID = getEnvOrDefault("BUCKET_ACCESS_KEY", "minioadmin")
+	useSSL = getEnvOrDefault("BUCKET_SSL", "false") == "true"
```
Comment (kirederik) [inline test/system/system_suite_test.go:77]:
```suggestion
	useSSL = os.GetEnv("BUCKET_SSL") == "true"
```

File: test/system/system_suite_test.go [hunk 3]
```diff
@@ -29,17 +28,63 @@ func TestSystem(t *testing.T) {
 
 var _ = SynchronizedBeforeSuite(func() {
 	//this runs once for the whole suite
+	worker = &destination{
+		context:       getEnvOrDefault("WORKER_CONTEXT", "kind-worker"),
+		checkExitCode: true,
```
Comment (kirederik) [inline test/system/system_suite_test.go:33]: nit: this seems to be the only value we care about? maybe `ignoreExitCode` which defaults to `false` would be better?

File: test/system/system_suite_test.go (hunk 2 above)
Comment (aclevername) [reply test/system/system_suite_test.go:77]: this is the most derik comment ever

File: test/system/system_suite_test.go (hunk 3 above)
Comment (aclevername) [reply test/system/system_suite_test.go:33]: sounds good! I'll refactor

File: Makefile [hunk 4]
```diff
@@ -216,6 +216,14 @@ build-and-load-bash: # Build and load all test pipeline images
 	kind load docker-image syntassodev/bash-promise:v1alpha1 --name platform
 	kind load docker-image syntassodev/bash-promise:v1alpha2 --name platform
 
+build-and-push-bash:
```
[the remark this answers is not in the document]
Comment (aclevername) [reply Makefile:219]: one day, but not today. Tbf the same outcome is probably acheiveable by env vars instead, but i cba to do that now

File: test/system/system_suite_test.go (hunk 2 above)
Comment (kirederik) [reply test/system/system_suite_test.go:73]: its called "writing good code" @aclevername

File: test/system/assets/bash-promise/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    kratix-promie-release: pod
```
Comment (SaphMB) [inline test/system/assets/bash-promise/deployment.yaml:6]:
```suggestion
    kratix-promise-release: pod
```

File: test/system/system_suite_test.go [hunk 6]
```diff
@@ -29,17 +28,59 @@ func TestSystem(t *testing.T) {
...
 	platform.kubectl("apply", "-f", "../../hack/destination/gitops-tk-install.yaml")
+	platform.kubectl("apply", "-f", "./assets/bash-promise/deployment.yaml")
 	platform.kubectl("apply", "-f", catAndReplaceFluxResources(tmpDir, "./assets/git/platform_gitops-tk-resources.yaml"))
 	platform.kubectl("apply", "-f", catAndReplaceFluxResources(tmpDir, "./assets/git/platform_kratix_destination.yaml"))
 	os.RemoveAll(tmpDir)
-}, func() {})
+}, func() {
+	//this runs before each test
+
+	//This variable gets set in func above, but only for 1 of the nodes, so we set
```
Comment (SaphMB) [inline test/system/system_suite_test.go:59]:
```suggestion
	//These variables get set in func above, but only for 1 of the nodes, so we set
	//them again here to ensure all nodes have them
```
Assuming this is in reference to both the `worker` and `platform` variables

File: test/system/system_test.go [hunk 7]
```diff
@@ -31,9 +28,10 @@ import (
 )
 
 type destination struct {
-	context       string
-	checkExitCode bool
-	exitCode      int
+	context         string
+	ignoreExistCode bool
```
Comment (SaphMB) [inline test/system/system_test.go:32]:
```suggestion
	ignoreExitCode bool
```

File: test/system/system_test.go [hunk 8]
```diff
@@ -736,58 +728,73 @@ func requestWithNameAndCommand(name string, containerCmds ...string) string {
...
 func (c destination) kubectl(args ...string) string {
-	args = append(args, c.context)
+	args = append(args, "--context="+c.context)
 	command := exec.Command("kubectl", args...)
 	session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
 	ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
-	if c.checkExitCode {
-		EventuallyWithOffset(1, session, timeout, interval).Should(gexec.Exit(0))
-	} else {
+	if c.ignoreExistCode {
```
Comment (SaphMB) [inline test/system/system_test.go:779]:
```suggestion
	if c.ignoreExitCode {
```

File: test/system/system_test.go [hunk 9]
```diff
@@ -736,58 +728,73 @@ func requestWithNameAndCommand(name string, containerCmds ...string) string {
...
 }
 
 func (c destination) clone() destination {
 	return destination{
-		context:       c.context,
-		exitCode:      c.exitCode,
-		checkExitCode: c.checkExitCode,
+		context:         c.context,
+		exitCode:        c.exitCode,
+		ignoreExistCode: c.ignoreExistCode,
```
Comment (SaphMB) [inline test/system/system_test.go:791]:
```suggestion
		ignoreExitCode: c.ignoreExitCode,
```

File: test/system/system_test.go [hunk 10]
```diff
@@ -736,58 +728,73 @@ func requestWithNameAndCommand(name string, containerCmds ...string) string {
...
 	}
 }
 
 func (c destination) ignoreExitCode() destination {
 	newDestination := c.clone()
-	newDestination.checkExitCode = false
+	newDestination.ignoreExistCode = true
```
Comment (SaphMB) [inline test/system/system_test.go:797]:
```suggestion
	newDestination.ignoreExitCode = true
```

**Review by SaphMB — CHANGES_REQUESTED** [review]
Nothing to add other than some typo corrections

**Pull request merged**